src/lib.rs

```diff
@@ -189,3 +219,238 @@ impl From<Options> for original::Options {
         options
     }
 }
+
+impl PartialEq<std::ops::Range<usize>> for pulldown::Range {
+    fn eq(&self, other: &std::ops::Range<usize>) -> bool {
+        self.start == other.start.try_into().unwrap() && self.end == other.end.try_into().unwrap()
+    }
+}
+
+impl PartialEq<original::Event<'_>> for Event {
+    fn eq(&self, other: &original::Event<'_>) -> bool {
+        match (self, other) {
+            (Self::Start(l0), original::Event::Start(r0)) => l0 == r0,
+            (Self::End(l0), original::Event::End(r0)) => l0 == r0,
+            (Self::Text(l0), original::Event::Text(r0)) => l0 == &r0.clone().into_string(),
+            (Self::Code(l0), original::Event::Code(r0)) => l0 == &r0.clone().into_string(),
+            (Self::Html(l0), original::Event::Html(r0)) => l0 == &r0.clone().into_string(),
+            (Self::FootnoteReference(l0), original::Event::FootnoteReference(r0)) => {
+                l0 == &r0.clone().into_string()
+            }
+            (Self::SoftBreak, original::Event::SoftBreak) => true,
+            (Self::HardBreak, original::Event::HardBreak) => true,
+            (Self::Rule, original::Event::Rule) => true,
+            (Self::TaskListMarker(l0), original::Event::TaskListMarker(r0)) => l0 == r0,
+            _ => false,
+        }
+    }
+}
+
+impl PartialEq<original::Tag<'_>> for Tag {
+    fn eq(&self, other: &original::Tag<'_>) -> bool {
+        match (self, other) {
+            (Self::Paragraph, original::Tag::Paragraph) => true,
+            (Self::Heading(l0), original::Tag::Heading(r0, r1, r2)) => {
+                l0.0 == *r0
+                    && l0.1 == r1.map(str::to_string)
+                    && l0.2 == *r2.iter().map(|&s| String::from(s)).collect::<Vec<String>>()
```

Comment:
You should be able to replace this with `Iterator::eq()` or `Iterator::eq_by()`:

```suggestion
                    && l0.2.iter().eq(r2)
```